core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -1155,11 +1155,12 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     val testKey = "hadoop.tmp.dir"
     val bufferKey = "io.file.buffer.size"
     val hadoopConf0 = new Configuration()
+    hadoopConf0.set(testKey, "/tmp/hive_zero")
 
     val hiveConfFile = Utils.getContextOrSparkClassLoader.getResource("hive-site.xml")
     assert(hiveConfFile != null)
     hadoopConf0.addResource(hiveConfFile)
-    assert(hadoopConf0.get(testKey) === "/tmp/hive_one")
+    assert(hadoopConf0.get(testKey) === "/tmp/hive_zero")
```

Comment:
This is okay because Apache Spark UT aims to test line 1170 ~ 1172.
```
    assert(sc.hadoopConfiguration.get(testKey) === "/tmp/hive_one",
      "hive configs have higher priority than hadoop ones ")
    assert(sc.hadoopConfiguration.get(bufferKey).toInt === 65536,
      "spark configs have higher priority than hive ones")
```

Reply:
w/ this change, `"/tmp/hive_one"` in `hive-site.xml` overlays `"/tmp/hive_{user}"` not  `"/tmp/hive_zero"`, which happens in `SparkHadoopUtil.appendS3AndSparkHadoopHiveConfigurations`.

As `hadoopConf0` is not passed to `SparkHadoopUtil`, while `core-site.xml` is, so here to reduce test flakiness, we can remove those `assert`s for `hadoopConf0`, as they are just used to check `hive-site.xml` overrides `core-site.xml`.

Reply:
@yaooqinn, can you open a PR for that, or push some changes into this PR? I think the test is not flaky but broken. It doesn't look obvious because we don't always run YarnClusterSuite.